pkg/rbd/nodeserver.go

```diff
@@ -103,14 +135,6 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	targetPathMutex.LockKey(targetPath)
 	defer targetPathMutex.UnlockKey(targetPath)
 
-	notMnt, err := ns.mounter.IsLikelyNotMountPoint(targetPath)
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
-	}
-	if notMnt {
-		return nil, status.Error(codes.NotFound, "Volume not mounted")
-	}
-
 	devicePath, cnt, err := mount.GetDeviceNameFromMount(ns.mounter, targetPath)
```

Comment:
@rootfs

As a result of discussion in CSI WG, csi driver needs bind mount device to target path for block volume case.  

However, on unpublish, above code doesn't seem to return a proper devicePath for block volume.
Because mount command returns ```devtmpfs``` instead of devicePath, if the block volume is bind mounted.
On the other hand, ```findmnt``` will return an information on devicePath.

Are there any good ways to utilize ```findmnt``` command for this purpose inside csi driver, here?
 

```
# touch /tmp/dev
# mount -o bind /dev/rbd0 /tmp/dev 
# mount | grep /tmp/dev 
devtmpfs on /tmp/dev type devtmpfs (rw,nosuid,seclabel,size=16460544k,nr_inodes=4115136,mode=755)

# findmnt -n -o SOURCE --target /tmp/dev  
devtmpfs[/rbd0]
```

Reply:
in this case, you can use the block's major and minor number to find the device path. 
https://www.kernel.org/doc/Documentation/admin-guide/devices.txt

Reply:
```console
#mount -o bind /dev/sdc /tmp/dev

# ls -l /tmp/dev
brw-rw----. 1 root disk 8, 32 Nov  2 09:36 /tmp/dev

# so the major number is 8, minor number is 32, you can also get them from stat

# then find the same major and minor number in /dev
# ls -l /dev/sd*
brw-rw----. 1 root disk 8,  0 Oct  8 19:00 /dev/sda
brw-rw----. 1 root disk 8,  1 Oct  8 19:00 /dev/sda1
brw-rw----. 1 root disk 8,  2 Oct  8 19:00 /dev/sda2
brw-rw----. 1 root disk 8, 16 Oct  8 19:00 /dev/sdb
brw-rw----. 1 root disk 8, 17 Oct  8 19:00 /dev/sdb1
brw-rw----. 1 root disk 8, 18 Oct  8 19:00 /dev/sdb2
brw-rw----. 1 root disk 8, 32 Nov  2 09:36 /dev/sdc
```